src/app/core/rest/collection.service.ts

```diff
@@ -43,13 +43,14 @@ export class CollectionService extends PermissionService {
     super(httpClient, store);
   }
 
-  public createCollection(collection: Collection): Observable<string> {
+  public createCollection(collection: Collection): Observable<Collection> {
     return this.httpClient.post(
       this.apiPrefix(), this.toDto(collection),
       {observe: 'response', responseType: 'text'}
     ).pipe(
       map(response => response.headers.get('Location').split('/').pop()),
       tap(code => this.homePageService.addLastUsedCollection(code).subscribe()),
+      map(code => ({...collection, code: code})), // TODO return fresh instance from the server instead
```

Comment:
In some of previous commits you told me not to do that (another request to server)... why it's now okay? (reaction to comment)

Reply:
Do you mean this spread syntax (`{...something}`)? It depends what you want to accomplish. In this case, when you call `createCollection` method and pass your `collection` as a parameter, its `code` will not be modified. But when you for example modify some of the attributes in the returned collection, they will be modified in the collection passed as a parameter since they use the same inner objects.

When using NgRx (or Redux), you try to make all of your values immutable. If you want to modify something, you make a copy of it and change the value in a new object. If you want to modify inner object, you make copy of that one as well. It means that there will be many cases when the inner objects are actually shared between multiple states but it does not matter since you are not allowed to modify them anyway.

Reply:
Thank you for explanation, but my answer was to heading to TODO comment after the line (return fresh instance from server)... But maybe it's because NgRx, isn't it?

Reply:
Oh, I see. This is about changing the back-end in order to return the instance of the newly created collection instead of getting the collection in another request as you did.